### PR TITLE
Bot language clarification

### LINKED ISSUE #6469: Improve k8s-ci-bot language to be more clear
In k/test-infra's [contributing](https://github.com/kubernetes/test-infra/blob/master/CONTRIBUTING.md) instructions, the administrative requirements for having your pull request approved and lgtm'd is described fairly clearly.

The actual bot prompts should outline these requirements and remaining steps to take more clearly, with the contributor's level of authorization in mind.

Currently, if a non-owner submits a pull request, the bot response will say:

"[APPROVALNOTIFIER] This PR is NOT APPROVED"

followed by

"This pull-request has been approved by: <author>"

followed by instructions for code owners (not the reviewer).

As the non-owner author, I see "NOT APPROVED" followed by "approved by <me>" followed by "you can assign people" (feels like I'm intruding on someone's time) and "you can /approve as follows" (which is not really meant for me, the author).

### PR DESCRIPTION
**Pull request by guineveresaenger**
Adds clarifying language to bot message on pull requests.
Fixes #6469 

No actual code logic was changed; this is a first attempt to have the bot logically explain the "lgtm" vs "approve" workflow. I am open to suggestions on changing the logic itself.

### CODE REVIEW

File: prow/plugins/approve/approvers/owners.go [hunk 1]
```diff
@@ -591,8 +593,9 @@ The full list of commands accepted by this bot can be found [here](https://go.k8
 Needs approval from an approver in each of these OWNERS Files:
 
 {{range .ap.GetFiles .org .project}}{{.}}{{end}}
-You can indicate your approval by writing `+"`/approve`"+` in a comment
-You can cancel your approval by writing `+"`/approve cancel`"+` in a comment
+OWNER approvers can indicate their approval by writing `+"`/approve`"+` in a comment
+OWNER approvers can cancel approval by writing `+"`/approve cancel`"+` in a comment
+A PR needs both /approve and /lgtm to be merged by the bot. An OWNER approver's /lgtm automatically implies an /approve.
```
Comment (stevekuznetsov) [inline prow/plugins/approve/approvers/owners.go]: Whether or not `/approve` and/or `/lgtm` are required to merge depends on how the merge is configured (in `mungegithub` and/or `tide`)

File: prow/plugins/approve/approvers/owners.go [hunk 2]
```diff
@@ -591,8 +593,9 @@ The full list of commands accepted by this bot can be found [here](https://go.k8
 Needs approval from an approver in each of these OWNERS Files:
 
 {{range .ap.GetFiles .org .project}}{{.}}{{end}}
-You can indicate your approval by writing `+"`/approve`"+` in a comment
-You can cancel your approval by writing `+"`/approve cancel`"+` in a comment
+OWNER approvers can indicate their approval by writing `+"`/approve`"+` in a comment
```
Comment (stevekuznetsov) [inline prow/plugins/approve/approvers/owners.go]: Maybe `Approvers can...` ? The `OWNER approvers` phrase is a little confusing to me

Comment (guineveresaenger) [reply prow/plugins/approve/approvers/owners.go]: The problem is that "approve" means two differnet things in this context. :/

File: prow/plugins/approve/approvers/owners.go (hunk 1 above)
Comment (guineveresaenger) [reply prow/plugins/approve/approvers/owners.go]: I was under the impression that a PR needs the following labels
1. "lgtm"
2. "approve"
with the additional confusion thrown in that lgtm from an approver (from the owners file) is an automatic approve.

Comment (guineveresaenger) [reply prow/plugins/approve/approvers/owners.go]: Can you point me to the place where the merge is configured? I am happy to change the logic, but at this point it's unclear to me how the logic flows.

Comment (stevekuznetsov) [reply prow/plugins/approve/approvers/owners.go]: In the most common configuration, that is true, but there is no hard requirement for that to be the case. Not sure how we can make this comment more clear as the configuration for the merge behavior is not available to this plugin.

Comment (guineveresaenger) [reply prow/plugins/approve/approvers/owners.go]: I see. Thank you for explaining. I'll go with your suggestion.

File: prow/plugins/approve/approvers/owners.go (hunk 2 above)
Comment (cjwagner) [reply prow/plugins/approve/approvers/owners.go]: How about `Approvers listed in OWNERS files...`?

File: prow/plugins/approve/approvers/owners.go [hunk 3]
```diff
@@ -565,6 +565,8 @@ Approval requirements bypassed by manually added approval.
 This pull-request has been approved by: {{range $index, $approval := .ap.ListApprovals}}{{if $index}}, {{end}}{{$approval}}{{end}}
 
 {{- if (and (not .ap.AreFilesApproved) (not (call .ap.ManuallyApproved))) }}
+To fully approve this pull request, please assign additional approvers. Any member of the organization can review and `+"`/lgtm`"+` a pull request, 
+and can be assigned to this PR, but in order for the pull request to be merged, it additionally has to be approved by a code owner listed in the OWNERS file for this repository(details below).
```
Comment (cjwagner) [inline prow/plugins/approve/approvers/owners.go]: This is text shouldn't be specific to our merge requirements. It should only describe the approval process which is just one of our merge requirements. The first sentence is fine, but `lgtm` may not always be required for merge.

**Comment by guineveresaenger** [comment]
I added in a link that explains the PR approval process. I think it's worth it to leave some bread crumbs for confused contributors, but I understand if we want to be conservative about sending links to other repositories. Let me know what you think!

**Pull request merged**